documents/validator.go

```diff
@@ -250,17 +250,20 @@ func readyForSignaturesValidator(didBytes, priv, pub []byte) Validator {
 			return errors.New("expecting only one signature")
 		}
 
-		s := crypto.Sign(didBytes, priv, pub, cd.SigningRoot)
+		s, err := crypto.SignMessage(priv, cd.SigningRoot, crypto.CurveSecp256K1, true)
```

Comment:
I don't understand why we actually need to sign here to verify...  What does this check do?

Reply:
This check is triggered when an update/create in a document and before the request signatures from peers is triggered. It checks that the signature[0] "creator" is valid. I am not sure if this is completely needed. In the case that we want a precheck at that stage, we should check all the signatures in the document not only the first one.

Reply:
A verify on each of them instead of sign and validate.